contributors/design-proposals/node/cache-management.md

```diff
@@ -0,0 +1,220 @@
+pinning. The underlying cache management library provides such support, but in
+this experimental  release, we opt for overlapping pools because of the greater
+sharing they provide (leading to better best-effort performance) yet providing
+less contention for workloads that have a higher QOS requirement. Further, when
+workloads are pinned to a CPU core, they already have exclusive access to the
+core’s L1 and L2 cache.
+
+### Pre-defined COS
+Out of the box, there will be a default set of COS with their associated cache
+ways. Documentation shall be provided to indicate how to alter them.
+
+1. SHARED - includes only the lower half of the cache ways
+1. RESERVED that includes all the cache ways
+
+In the presence of cache allocation functionality on a platform, Kube
+initialization code that runs when a new host is added to a cluster will
+register the default or otherwise configured COS.  All Kube and System daemon
+workloads could be assigned to either RESERVED or SHARED, and is configurable. 
+
+## Example Scenarios
+
+1. A Container arrives with Guaranteed QoS and other cpu and RAM specific 
+   requirements
+   1. Kuberuntime calls the CRI delegate to create the container
+   1. Kuberuntime adds the containers with the CPU Manager
+   1. CPU Manager adds the container to the static policy. The container is
+      added to the RESERVED (in terms of number of cache ways) COS.
+   1. No change flows related to all container state-reads and resource updates
+      that today use the CRI.
+1. A Container arrives with Non-Guraranteed QoS and other cpu and RAM specific
+   requirements
+   Everything proceeds as in Case 1 above except for step ( c ) where the
+   container is assigned to the SHARED by virtue of the number of cache ways it
+   includes.
+1. A container exits
+   Everything proceeds as with other container handling by the CPU manager with
+   an additional step, the container to COS mapping is cleared and updates via
+   CRI reflect the same.
+
+## Evaluation
+
+The following checklist has to be evaluated.
+
+- No perceptible performance difference when the policy is “None” in container
+  performance
+- No perceptible performance difference when the policy is “Static” or
+  “Dynamic” and all container workloads submitted have a priority equal to high.
+- Perceptible performance improvement when a latency/jitter sensitive container
+  workload is assigned a priority equal to high and all other workloads are
+  assigned priority low with either “Static” or “Dynamic” policy.
+- Perceptible difference in performance, all things being equal with respect to
+  launched containers except for a streaming application such as a video player
+  running first as  low priority and next as high priority. In the former
+  scenarios, cache disruption is limited to the bottom half while in the latter
+  it has full use of the cache and thus affects adversely more co-resident
+  container workloads. 
+
+## Implementation Roadmap
```

Comment:
I'd like to see monitoring of cache allocations and memory bandwidth prior to adding a feature to restrict it. How can a k8s user know if they need to consume cache partitioning?

Reply:
True. We are working on another proposal to integrate cache monitoring technology with cadvisor, which will allow user to monitor the cache usage of the specific container.